x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/pivot/AggregationResultUtils.java

```diff
@@ -102,6 +84,23 @@
         });
     }
 
+    static AggValueExtractor getExtractor(Aggregation aggregation, String fieldType) {
+        if (aggregation instanceof NumericMetricsAggregation.SingleValue) {
+            return new SingleValueAggExtractor((SingleValue) aggregation, fieldType);
```

Comment:
_opinion_: 

this looks definitely nicer, but now we are creating 1 more object for every bucket result.

The extractor object is created and `extractor.value()` gets immediately called on the next line (line 76, 77). So the object is very short-lived. It should be possible to return static instances and pass `aggregation` and `fieldType` as part of `value()`.

Reply:
@hendrikmuhs good idea, a static factory. We could still get the benefit of inherited overrides but keep the memory footprint lower.